client/src/lib/ipfs/quicknodeIpfsClient.ts

```diff
@@ -0,0 +1,116 @@
+import axios from 'axios';
+import FormData from 'form-data';
+import { IpfsClient } from "./ipfsClient";
+
+export class QuicknodeIpfsClient extends IpfsClient {
+  private gatewayUrl: string;
+  private apiKey: string;
+
+  constructor(apiKey: string, gatewayUrl: string) {
+    super("Quicknode");
+    if (gatewayUrl[gatewayUrl.length - 1] === "/") {
+      gatewayUrl = gatewayUrl.slice(0, -1);
+      if (!gatewayUrl.endsWith("/ipfs")) {
+        gatewayUrl += "/ipfs";
+      }
+    }
+    this.gatewayUrl = gatewayUrl;
+    this.apiKey = apiKey;
+  }
+
+  async getSize(hashOrCid: string): Promise<number | null> {
+    try {
+      if (hashOrCid.startsWith("0x")) {
+        hashOrCid = this.convertBytes32ToIpfsCid(hashOrCid);
+      }
+      const res = await axios.head(`${this.gatewayUrl}/${hashOrCid}`);
+      return res.headers["content-length"] ?? null;
+    } catch (e: any) {
+      console.error(e?.message);
```

Comment:
We will use this client in backend so `console.error` will page oncalls. We should let the caller to decide logging.
It hides the error details from the caller. The caller cannot know if the transient error or not. We should let ether:
1. `IPFSClient` own retrying and provide `maxRetry` to constructor. Or
2. Throw an error which can distinguish transient errors or unrecoverable.

Reply:
after some discussion with Xinding offline this will be part of a broader retry logic change in the future: https://linear.app/intrinsictech/issue/INT-1031/add-retry-logic-in-client

Reply:
I thought we said we would **not** use the client in backend? This is to remove the dependency on this big package from the backend

Reply:
The logic is generic. Developers will need it

Reply:
Discussed offline, we can move the ipfs client implementation to `axiom-tools`